modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/service/test/KaleoInstanceLocalServiceTest.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.workflow.kaleo.service.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.blogs.model.BlogsEntry;
+import com.liferay.blogs.service.BlogsEntryLocalService;
+import com.liferay.portal.kernel.exception.NoSuchWorkflowInstanceLinkException;
+import com.liferay.portal.kernel.service.WorkflowDefinitionLinkLocalService;
+import com.liferay.portal.kernel.service.WorkflowInstanceLinkLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.workflow.kaleo.service.KaleoInstanceLocalService;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Mateus Xavier
+ */
+@RunWith(Arquillian.class)
+public class KaleoInstanceLocalServiceTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testDeleteKaleoInstance() throws Exception {
+		_workflowDefinitionLinkLocalService.updateWorkflowDefinitionLink(
+			TestPropsValues.getUserId(), TestPropsValues.getCompanyId(), 0,
+			BlogsEntry.class.getName(), 0, 0, "Single Approver", 1);
+
+		BlogsEntry blogsEntry = _blogsEntryLocalService.addEntry(
+			TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(),
+			ServiceContextTestUtil.getServiceContext(
+				TestPropsValues.getGroupId()));
+
+		Assert.assertNotNull(
+			_workflowInstanceLinkLocalService.getWorkflowInstanceLink(
+				blogsEntry.getCompanyId(), blogsEntry.getGroupId(),
+				BlogsEntry.class.getName(), blogsEntry.getEntryId()));
+
+		_kaleoInstanceLocalService.deleteKaleoInstance(
+			_workflowInstanceLinkLocalService.getWorkflowInstanceLink(
+				blogsEntry.getCompanyId(), blogsEntry.getGroupId(),
+				BlogsEntry.class.getName(), blogsEntry.getEntryId()
+			).getWorkflowInstanceId());
+
+		Assert.assertThrows(
+			NoSuchWorkflowInstanceLinkException.class,
+			() -> _workflowInstanceLinkLocalService.getWorkflowInstanceLink(
+				blogsEntry.getCompanyId(), blogsEntry.getGroupId(),
+				BlogsEntry.class.getName(), blogsEntry.getEntryId()
+			).getWorkflowInstanceLinkId());
```

Comment:
@mxs2 we don't need to invoke the method `.getWorkflowInstanceLinkId()`, please remove this call.

Thanks!